File: domain/src/main/java/es/santirivera/surveilfall/domain/usecases/base/UseCaseTaggedCallback.java
package es.santirivera.surveilfall.domain.usecases.base;

public interface UseCaseTaggedCallback<Result, Err extends StringErrorOutput> {

	boolean isReady();

	void onSuccess(String tag, Result response);

	void onError(String tag, Err error);

	void onGenericError(String tag);

	void onNetworkUnavailable(String tag);
}

File: domain/src/main/java/es/santirivera/surveilfall/domain/usecases/base/UseCaseTaggedCallback.kt
package es.santirivera.surveilfall.domain.usecases.base

interface UseCaseTaggedCallback<Result, Err : StringErrorOutput> {

    fun onSuccess(tag: String, response: Result)

    fun onError(tag: String, error: Err)

    fun onGenericError(tag: String)

    fun onNetworkUnavailable(tag: String)
}
